onnxruntime/python/tools/quantization/qdq_quantizer.py

```diff
@@ -181,137 +203,182 @@ def try_replacing_upstream_output(self, upstream_output_name, output_name):
+                dq_output,
+                add_dequant_suffix(tensor_name),
+                scale_name,
+                zp_name,
+            )
+
+            quantized_value = QuantizedValue(
+                tensor_name,
+                dq_output,
+                scale_name,
+                zp_name,
+                QuantizedValueType.Input,
+            )
+            self.quantized_value_map[tensor_name] = quantized_value
+
+    def _quantize_normal_tensors(self):
+        for tensor_name, tensor_info in self.tensors_to_quantize.copy().items():
+            if tensor_name in self.quantized_value_map.keys():
+                continue
+
+            if tensor_info.tensor_type == QDQQuantTensorType.NORMAL:
+                # Quantize the input
+                initializer = find_by_name(tensor_name, self.model.initializer())
+                if initializer:
+                    self._add_qdq_pair_for_weight(initializer, tensor_info.axis)
                 else:
-                    q_input = tensor_name
-                    q_output = tensor_name + "_QuantizeLinear"
-                    dq_input = q_output
-                    dq_output = tensor_name + "_DequantizeLinear"
-                    if self.model.is_graph_output(tensor_name):
-                        q_input = tensor_name + "_QuantizeLinearInput"
-                        dq_output = tensor_name
-                        self.model.replace_output_of_all_nodes(tensor_name, q_input)
+                    data_found, scale_name, zp_name, _, _ = self._get_quantization_params(tensor_name)
+
+                    if data_found == False:
+                        raise ValueError(
+                            "Quantization parameters are not specified for param {}. "
+                            "In static mode quantization params for inputs and outputs of nodes to be quantized are required.".format(
+                                tensor_name
+                            )
+                        )
+
+                    self._add_qdq_pair_for_activation(tensor_name, scale_name, zp_name)
+
+                del self.tensors_to_quantize[tensor_name]
+
+    def _quantize_sharing_param_tensors(self):
+        while self.tensors_to_quantize:
+            has_update = False
+            for tensor_name, tensor_info in self.tensors_to_quantize.copy().items():
+                tensor_provider_name = tensor_info.quant_para_provider
+                if tensor_provider_name in self.quantized_value_map:
+                    del self.tensors_to_quantize[tensor_name]
+
+                    quantized_value = self.quantized_value_map[tensor_provider_name]
+                    # Quantize the input
+                    initializer = find_by_name(tensor_name, self.model.initializer())
+                    if initializer is not None:
+                        raise ValueError("Quantization parameter shared mode is not supported for weight yet")
                     else:
```

Comment:
R1720: Unnecessary "else" after "raise", remove the "else" and de-indent the code inside it (no-else-raise)
Raw output